### DISCUSSION TITLE
Potential Resource Exhaustion vulnerability in Cronicle's use of Glob

### DISCUSSION THREAD

**Discussion by matthewjhands** [jhuckaby/Cronicle#837] [Ideas]
Hello,

The SCA dependency scanning tool where I work and use Cronicle has picked up a "high" vulnerability in the `glob` dependency of Cronicle, because `glob@5.0.15` itself has a dependency on `inflight@1.0.6` which has a known memory leak issue, ⁠[which at least in theory could lead to a resource exhaustion vulnerability.](https://security.snyk.io/vuln/SNYK-JS-INFLIGHT-6095116) The Inflight maintainers have indicated that they won't be attempting to fix this bug ⁠[because the whole project is deprecated](https://github.com/isaacs/inflight-DEPRECATED-DO-NOT-USE/issues/5#issuecomment-2126153820).

I have a question and a feature request:

1. Do you think there is any realistic resource exhaustion vulnerability to Cronicle, given how it uses `glob`?
1. Is there any chance that Cronicle could be upgraded to use `glob@9.x`, ⁠[which apparently doesn't leverage inflight](https://github.com/isaacs/inflight-DEPRECATED-DO-NOT-USE/issues/5#issuecomment-1876974883) please?

Many thanks,
Matt

**Comment by jhuckaby** [comment]
Hi there!  

How weird, NPM audit doesn't catch that vuln at all, nor did I receive a notification from my snyk.io account 🤷🏻‍♂️

Oh well, I just went ahead and removed glob as a dependency entirely.  Cronicle v0.9.63 now uses my own glob implementation in [pixl-tools](https://github.com/jhuckaby/pixl-tools) (which uses [picomatch](https://www.npmjs.com/package/picomatch) under the hood).

Fixed in v0.9.63: https://github.com/jhuckaby/Cronicle/releases/tag/v0.9.63

**Reply by matthewjhands** [reply]
@jhuckaby you are the best. Thanks for getting this fixed so quickly (18 minutes!).

Many thanks and have a lovely Christmas break. 

Matt
